src/__tests__/02c-onchain-send.ts

```diff
@@ -170,8 +170,12 @@ it("Sends onchain payment _with memo", async () => {
   const { address } = await createChainAddress({ format: "p2wpkh", lnd: lndOutside1 })
   const paymentResult = await userWallet0.onChainPay({ address, amount, memo })
   expect(paymentResult).toBe(true)
-  const txs = await userWallet0.getTransactions()
-  expect((first(txs) as any).description).toBe(memo)
+  const txs: Record<string, string>[] = await userWallet0.getTransactions()
+  const firstTxs = first(txs)
+  if (!firstTxs) {
+    throw Error("No transactions found")
```

Comment:
Do we really need to throw an Error here? Isn't there a more elegant way to fail the test at this point? Or can move the check out of the test code into the production code and be explicit / safer with what type we are returning? (Same questions apply to the change higher up).

Reply:
my guess:

`first` can return an object or `null` if `firstTxs` array is empty. so because we exist the loop if `firstTxs` is null, then the now after this `if` that `firstTxs` is not null. 

my (bad) workaround, as highlighted below, was casting the result of the array as `any` (either with `any` or the use of `!`)

Reply:
I wish TypeScript would respect tests expectations, but it does not. I assumed the test HAS to have at least one transaction here, otherwise it should fail.

Reply:
Thats what I mean can't we encode this expectation in form of a test assertion `expect(firxtTxs).notTo(be(null))` or something like that.

Reply:
Yes, but unfortunately, TypeScript would still complain that's the variable might be undefined.